python/ccxt/bitget.py

```diff
@@ -4634,40 +4636,51 @@ def fetch_order(self, id: str, symbol: Str = None, params={}):
             market = self.market(symbol)
         request = {
             'orderId': id,
+            'symbol': market['id'],
         }
         response = None
-        order_type = self.safe_string(params, 'type', None)
+        order_type = params.pop('type', None)
         if market['spot']:
             if order_type == 'stop':
-                orderId = int(request.pop('orderId'))
-                request['isLessThan'] = str(orderId + 1)
-                request['limit'] = '1'
-                request['symbol'] = market['id']
-                response = self.privateSpotGetV2SpotTradeCurrentPlanOrder(request)
-                if isinstance(response, str):
-                    response = json.loads(response)
-                data = self.safe_value(response, 'data')
-                data = self.safe_value(data, 'orderList')
-                first = self.safe_value(data, 0, data)
-                return self.parse_order(first, market)
-            else:
-                response = self.privateSpotGetV2SpotTradeOrderInfo(self.extend(request, params))
+                fetch_func = None
+                try:
+                    data = self.get_trigger_sub_order(id, symbol)
+                    order_id = self.safe_value(data, 'orderId')
+                    if order_id and order_id != 'null':
+                        request['orderId'] = order_id
+                    else:
+                        fetch_func = self.fetch_canceled_and_closed_orders
+                except:
+                    fetch_func = self.fetch_open_orders
+                if fetch_func:
+                    orders = fetch_func(symbol, params={'stop':True,'idLessThan':int(id)+1, 'limit':1})
+                    if orders:
+                        order = orders[0]
+                        if order['id'] == id:
+                            return order
+                    raise OrderNotFound(f'order {id} not found')
+            response = self.privateSpotGetV2SpotTradeOrderInfo(self.extend(request, params))
```

Comment:
Didn't you "else" it on purpose? since we have the except there, I think it's better...